sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -32,7 +33,25 @@ public static HttpPipeline Build(ClientOptions options, params HttpPipelinePolic
         /// <param name="perRetryPolicies">Client provided per-retry policies.</param>
         /// <param name="responseClassifier">The client provided response classifier.</param>
         /// <returns>A new instance of <see cref="HttpPipeline"/></returns>
-        public static HttpPipeline Build(ClientOptions options, HttpPipelinePolicy[] perCallPolicies, HttpPipelinePolicy[] perRetryPolicies, ResponseClassifier responseClassifier)
+        public static HttpPipeline Build(
+            ClientOptions options,
+            HttpPipelinePolicy[] perCallPolicies,
+            HttpPipelinePolicy[] perRetryPolicies,
+            ResponseClassifier responseClassifier)
+        {
+            return Build(options, perCallPolicies, perRetryPolicies, responseClassifier, null);
+        }
+
+        /// <summary>
+        /// Creates an instance of <see cref="HttpPipeline"/> populated with default policies, customer provided policies from <paramref name="options"/> and client provided per call policies.
+        /// </summary>
+        /// <param name="options">The customer provided client options object.</param>
+        /// <param name="perCallPolicies">Client provided per-call policies.</param>
+        /// <param name="perRetryPolicies">Client provided per-retry policies.</param>
+        /// <param name="responseClassifier">The client provided response classifier.</param>
+        /// <param name="defaultTransportOptions">The customer provided transport options which will be applied to the default transport.</param>
+        /// <returns>A new instance of <see cref="HttpPipeline"/></returns>
+        public static HttpPipeline Build(ClientOptions options, HttpPipelinePolicy[] perCallPolicies, HttpPipelinePolicy[] perRetryPolicies, ResponseClassifier responseClassifier, HttpPipelineTransportOptions? defaultTransportOptions = null)
```

Comment:
Should this always return DisposableTransport?

Reply:
Can you say more about what advantage you see if we did?

Reply:
It's more explicit to consumers. When calling this method they always need to dispose the result. For shared clients, we can return no-op `DisposablePipeline`. This way there is less chance to get into a situation where client authors forget to dispose the pipeline.

Reply:
Although I agree, can we do this effectively without changing the return type on the method? Consumers would still need to know to cast it.

Reply:
Not sure I understand the question.